src/Raven.Client/Documents/Identity/AsyncHiLoIdGenerator.cs

```diff
@@ -94,6 +94,10 @@ public async Task<long> NextIdAsync()
                 //local range is exhausted , need to get a new range
                 var maybeNextTask = new Lazy<Task>(GetNextRangeAsync);
 
+                // other thread got new range
+                if (id <= Range.Max)
+                    continue;
+
```

Comment:
If the thread is blocked here, you still got the race :-)

Reply:
done